src/script/signingprovider.h

```diff
@@ -74,6 +72,8 @@ class FillableSigningProvider : public SigningProvider
     void ImplicitlyLearnRelatedKeyScripts(const CPubKey& pubkey) EXCLUSIVE_LOCKS_REQUIRED(cs_KeyStore);
 
 public:
+    mutable CCriticalSection cs_KeyStore;
```

Comment:
why this move? is this preparatory?

Reply:
Yes. `cs_KeyStore` sometimes needs to be locked by the caller.